spec/actions/bump_version_update_changelog_create_pr_action_spec.rb

```diff
@@ -0,0 +1,108 @@
+describe Fastlane::Actions::BumpVersionUpdateChangelogCreatePrAction do
+  describe '#run' do
+    let(:mock_github_pr_token) { 'mock-github-pr-token' }
+    let(:mock_github_token) { 'mock-github-token' }
+    let(:mock_repo_name) { 'mock-repo-name' }
+    let(:mock_changelog_latest_path) { './fake-changelog-latest-path/CHANGELOG.latest.md' }
+    let(:mock_changelog_path) { './fake-changelog-path/CHANGELOG.md' }
+    let(:branch) { 'main' }
+    let(:editor) { 'vim' }
+    let(:auto_generated_changelog) { 'mock-auto-generated-changelog' }
+    let(:edited_changelog) { 'mock-edited-changelog' }
+    let(:current_version) { '1.12.0' }
+    let(:new_version) { '1.13.0' }
+
+    before(:each) do
+      allow(FastlaneCore::UI).to receive(:input).with('New version number: ').and_return(new_version)
+      allow(File).to receive(:read).with(mock_changelog_latest_path).and_return(edited_changelog)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:validate_local_config_status_for_bump).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:auto_generate_changelog).with(anything).and_return(auto_generated_changelog)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:edit_changelog).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:create_new_release_branch).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:replace_version_number).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:attach_changelog_to_master).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:commmit_changes_and_push_current_branch).with(anything)
+      allow(Fastlane::Helper::RevenuecatInternalHelper).to receive(:create_release_pr).with(anything)
+    end
+
+    it 'calls all the appropriate methods with appropriate parameters' do
```

Comment:
Whoa even with tests? 🤯

Reply:
IMO, this is the main reason for moving this into a plugin ❤️

Reply:
Yeah this is dope